copy/src/fork.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright © 2019-2020 The Radicle Foundation <hello@radicle.foundation>
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{convert::TryFrom, ffi, path::PathBuf};
+
+use either::Either;
+
+use librad::{
+    git::{
+        identities::{self, Person, Project},
+        local::{transport::CanOpenStorage, url::LocalUrl},
+        types::{
+            remote::{LocalPushspec, Remote},
+            Flat,
+            Force,
+            GenericRef,
+            Reference,
+            Refspec,
+        },
+        Urn,
+    },
+    git_ext::{self, OneLevel, Qualified, RefLike},
+    peer::PeerId,
+    refspec_pattern,
+};
+
+use crate::git;
+
+/// When checking out a working copy, we can run into several I/O failures.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// Git error when checking out the project.
+    #[error(transparent)]
+    Git(#[from] git::Error),
+
+    #[error(transparent)]
+    Identities(#[from] identities::Error),
+
+    #[error("the project, at `{0}`, does not have default branch set")]
+    MissingDefaultBranch(Urn),
+
+    #[error(transparent)]
+    Ref(#[from] git_ext::name::Error),
+
+    /// An error occurred in the local transport.
+    #[error(transparent)]
+    Transport(#[from] librad::git::local::transport::Error),
+}
+
+/// Based off of the `Fork`, clone the project using the provided inputs.
```

Comment:
Is the terminology "fork" used in `upstream`? That'd be unfortunate, because the defining property of a fork is a change of ownership, which is not what is happening here. Can we agonise over this?

Reply:
It's a mixture between "checkout" and "fork" :grimacing: If you're creating a copy from your own project it's a checkout and if you're transitioning from a project you're only tracking then it's a "fork".

Reply:
I'd propose "chork", then